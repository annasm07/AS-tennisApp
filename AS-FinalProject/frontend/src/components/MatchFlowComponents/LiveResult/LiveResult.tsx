import React from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {Text, View, StyleSheet} from 'react-native';
import matchBoxStyles from '../../../theme/matchBoxTheme';
import globalStyles from '../../../theme/globalThemes';
import {
  updateMatch,
  updateGames,
  updateSets,
  updateMatchGames,
  updateMatchSets,
} from '../../../redux/actions/actionCreators';
import {checkEndSet, checkEndMatch} from '../../../utils/checkEndSet';

export default function LiveResult({navigation}: any) {
  const currentMatch = useSelector((store: any) => store.currentMatch);
  let tokens = useSelector((store: any) => store.tokens);
  let {points} = useSelector((store: any) => store.currentGamePoints);
  let currentSetGames = useSelector((store: any) => store.currentSetGames);
  let currentMatchSets = useSelector((store: any) => store.currentMatchSets);

  const dispatch = useDispatch();

  function finishMatch() {
    const p1Sets =
      currentMatch.flow?.sets[currentMatch.flow.sets.length - 1]?.p1 || 0;
    const p2Sets =
      currentMatch.flow?.sets[currentMatch.flow.sets.length - 1]?.p2 || 0;
    (p1Sets === 2 || p2Sets === 2) && navigation.navigate('Dashboard');
  }

  function getDateString(dateString: any) {
    const dateToDisplay = new Date(dateString).toLocaleDateString();
    return dateToDisplay;
  }

  function finishSet(playerWhoWon: String) {
    dispatch(updateSets(playerWhoWon));
    dispatch(updateMatchSets(currentSetGames, currentMatchSets));
    checkEndMatch(playerWhoWon, currentSetGames) && finishMatch();
  }

  function finishGame(playerWhoWon: String) {
    dispatch(updateGames(playerWhoWon));
    dispatch(updateMatchGames(currentSetGames, points));
    checkEndSet(playerWhoWon, currentSetGames) && finishSet(playerWhoWon);
    dispatch(updateMatch(tokens[0], currentMatch));
  }

  function renderPlayerResult(PLAYER: any, OtherPLAYER: any) {
    if (points[points.length - 1][PLAYER] > 40) {
      if (
        points[points.length - 1][PLAYER] >
        points[points.length - 1][OtherPLAYER]
      ) {
        if (
          Math.abs(
            points[points.length - 1][PLAYER] -
              points[points.length - 1][OtherPLAYER],
          ) >= 2
        ) {
          finishGame(PLAYER);
          return <Text style={matchBoxStyles.playerPoint}>Game</Text>;
        } else {
          return <Text style={matchBoxStyles.playerPoint}>Ad</Text>;
        }
      } else if (
        points[points.length - 1][PLAYER] ===
        points[points.length - 1][OtherPLAYER]
      ) {
        return <Text style={matchBoxStyles.playerPoint}>40</Text>;
      } else if (
        points[points.length - 1][PLAYER] -
          points[points.length - 1][OtherPLAYER] ===
        2
      ) {
        finishGame(PLAYER);
        return <Text style={matchBoxStyles.playerPoint}>Game</Text>;
      } else {
        return <Text style={matchBoxStyles.playerPoint}>40</Text>;
      }
    } else {
      return (
        <Text style={matchBoxStyles.playerPoint}>
          {points[points.length - 1][PLAYER]}
        </Text>
      );
    }
  }

  return currentMatch.result ? (
    <View style={styles.view}>
      <View style={matchBoxStyles.matchBox}>
        <View style={matchBoxStyles.player1}>
          <Text>{currentMatch?.result[0]?.name}</Text>

          <View style={matchBoxStyles.playerResult}>
            {renderPlayerResult('p1', 'p2')}
            {currentMatch?.result[0]?.games.map((set: any) => (
              <Text> {set}</Text>
            ))}
          </View>
        </View>

        <View style={matchBoxStyles.player2}>
          <Text>{currentMatch?.result[1]?.name}</Text>

          <View style={matchBoxStyles.playerResult}>
            {renderPlayerResult('p2', 'p1')}
            {currentMatch?.result[1]?.games.map((set: any) => (
              <Text> {set}</Text>
            ))}
          </View>
        </View>
      </View>
      <View style={matchBoxStyles.dateBox}>
        <Text style={globalStyles.grayText}>
          {getDateString(currentMatch?.date)}
        </Text>
        <Text style={styles.time}>Best of 3 Sets</Text>
      </View>
    </View>
  ) : (
    <Text>...loading...</Text>
  );
}

const styles = StyleSheet.create({
  view: {
    height: 130,
  },
  time: {},
});
