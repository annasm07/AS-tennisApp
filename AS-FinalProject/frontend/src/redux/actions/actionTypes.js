export default {
  LOG_IN: 'LOG_IN',
  LOG_IN_ERROR: 'LOG_IN_ERROR',
  SIGN_UP: 'LOG_IN',
  SIGN_UP_ERROR: 'LOG_IN_ERROR',
  GET_PLAYER: 'GET_PLAYER',
  GET_PLAYER_ERROR: 'GET_PLAYER_ERROR',
  GET_PLAYERS: 'GET_PLAYERS',
  GET_PLAYERS_ERROR: 'GET_PLAYERS_ERROR',
  CREATE_MATCH: 'CREATE_MATCH',
  CREATE_MATCH_ERROR: 'CREATE_MATCH_ERROR',
  UPDATE_POINTS: 'UPDATE_POINTS',
  UPDATE_POINTS_ERROR: 'UPDATE_POINTS_ERROR',
  UPDATE_GAMES: 'UPDATE_GAMES',
  UPDATE_GAMES_ERROR: 'UPDATE_GAMES_ERROR',
  END_GAME: 'END_GAME',
  END_GAME_ERROR: 'END_GAME_ERROR',
  END_SET: 'END_SET',
  END_SET_ERROR: 'END_SET_ERROR',
  END_MATCH: 'END_MATCH',
  END_MATCH_ERROR: 'END_MATCH_ERROR',
};
