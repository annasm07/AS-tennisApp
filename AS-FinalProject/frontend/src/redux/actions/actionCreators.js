import axios from 'axios';
import actionTypes from './actionTypes';

const LOG_IN_URL = 'http://localhost:1616/api/auth/login';
const SIGN_UP_URL = 'http://localhost:1616/api/auth/signup';
const GET_PLAYER_URL = 'http://localhost:1616/api/home';
const GET_ALL_PLAYERS = 'http://localhost:1616/api/players';
const CREATE_MATCH_URL = 'http://localhost:1616/api/match';

export function logIn(email, password) {
  return async dispatch => {
    try {
      const {data} = await axios.post(LOG_IN_URL, {email, password});
      dispatch({
        type: actionTypes.LOG_IN,
        user: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.LOG_IN_ERROR,
      });
    }
  };
}

export function signUp({name, email, password, player, playerName}) {
  return async dispatch => {
    try {
      await axios.post(SIGN_UP_URL, {
        name,
        email,
        password,
        player,
        playerName,
      });
      dispatch({
        // type: actionTypes.SIGN_UP,
        // user: data.user,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.SIGN_UP_ERROR,
      });
    }
  };
}

export function getPlayerInfo(token, playerId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${GET_PLAYER_URL}/${playerId}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYER,
        player: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYER_ERROR,
      });
    }
  };
}
export function getAllPlayers(token) {
  return async dispatch => {
    try {
      const {data} = await axios(GET_ALL_PLAYERS, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYERS,
        players: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYERS_ERROR,
      });
    }
  };
}

export function newMatch(token, p1Name, p2Name) {
  return async dispatch => {
    try {
      const {data} = await axios.post(
        CREATE_MATCH_URL,
        {p1Name, p2Name},
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      );
      dispatch({
        type: actionTypes.CREATE_MATCH,
        currentMatch: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.CREATE_MATCH_ERROR,
      });
    }
  };
}
