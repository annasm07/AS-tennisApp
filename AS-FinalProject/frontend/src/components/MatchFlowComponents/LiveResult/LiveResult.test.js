import {render} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../../redux/actions/actionCreators';
import {Provider} from 'react-redux';
import LiveResult from './LiveResult';

jest.mock('../../../redux/actions/actionCreators');

describe('When invoked a LiveResult component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    currentMatch: {
      flow: {
        sets: [
          {
            p1: 0,
            p2: 1,
          },
          {
            p1: 1,
            p2: 1,
          },
        ],
      },
      result: [
        {
          games: [0, 0, 0],
          name: 'Anna',
        },
        {
          games: [0, 0, 0],
          name: 'Sala',
        },
      ],
    },
    currentGamePoints: {
      points: [
        {
          p1: 0,
          p2: 15,
        },
        {
          p1: 0,
          p2: 41,
        },
      ],
    },
    currentSetGames: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 1,
      },
    ],
    currentMatchSets: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 1,
      },
    ],
    server: true,
    tokens: ['abc', 'cde'],
    user: {user: {playerId: '123cde'}},
  });
  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'updateSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateServer').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatch').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });

  test('Should render LiveResult component', () => {
    const liveResult = render(
      <Provider store={store}>
        <LiveResult navigation={navigation} />
      </Provider>,
    );
    expect(liveResult).toMatchSnapshot();
  });
});

describe('When invoked a LiveResult component, with other values - game', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    currentMatch: {
      flow: {
        sets: [
          {
            p1: 0,
            p2: 1,
          },
          {
            p1: 1,
            p2: 1,
          },
        ],
      },
      result: [
        {
          games: [0, 0, 0],
          name: 'Anna',
        },
        {
          games: [0, 0, 0],
          name: 'Sala',
        },
      ],
    },
    currentGamePoints: {
      points: [
        {
          p1: 0,
          p2: 15,
        },
        {
          p1: 40,
          p2: 40,
        },
      ],
    },
    currentSetGames: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 6,
      },
    ],
    currentMatchSets: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 2,
      },
    ],
    server: true,
    tokens: ['abc', 'cde'],
    user: {user: {playerId: '123cde'}},
  });
  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'updateSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateServer').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatch').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });

  test('Should render LiveResult component', () => {
    const liveResult = render(
      <Provider store={store}>
        <LiveResult navigation={navigation} />
      </Provider>,
    );
    expect(liveResult).toMatchSnapshot();
  });
});

describe('When invoked a LiveResult component, with other values - game after ad', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    currentMatch: {
      flow: {
        sets: [
          {
            p1: 0,
            p2: 1,
          },
          {
            p1: 1,
            p2: 1,
          },
        ],
      },
      result: [
        {
          games: [0, 0, 0],
          name: 'Anna',
        },
        {
          games: [0, 0, 0],
          name: 'Sala',
        },
      ],
    },
    currentGamePoints: {
      points: [
        {
          p1: 0,
          p2: 15,
        },
        {
          p1: 41,
          p2: 43,
        },
      ],
    },
    currentSetGames: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 6,
      },
    ],
    currentMatchSets: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 2,
      },
    ],
    server: true,
    tokens: ['abc', 'cde'],
    user: {user: {playerId: '123cde'}},
  });
  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'updateSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateServer').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatch').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });

  test('Should render LiveResult component', () => {
    const liveResult = render(
      <Provider store={store}>
        <LiveResult navigation={navigation} />
      </Provider>,
    );
    expect(liveResult).toMatchSnapshot();
  });
});

describe('When invoked a LiveResult component, with other values - ad', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    currentMatch: {
      flow: {
        sets: [
          {
            p1: 0,
            p2: 1,
          },
          {
            p1: 1,
            p2: 1,
          },
        ],
      },
      result: [
        {
          games: [0, 0, 0],
          name: 'Anna',
        },
        {
          games: [0, 0, 0],
          name: 'Sala',
        },
      ],
    },
    currentGamePoints: {
      points: [
        {
          p1: 0,
          p2: 15,
        },
        {
          p1: 40,
          p2: 41,
        },
      ],
    },
    currentSetGames: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 6,
      },
    ],
    currentMatchSets: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 2,
      },
    ],
    server: true,
    tokens: ['abc', 'cde'],
    user: {user: {playerId: '123cde'}},
  });
  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'updateSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateServer').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatch').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });

  test('Should render LiveResult component', () => {
    const liveResult = render(
      <Provider store={store}>
        <LiveResult navigation={navigation} />
      </Provider>,
    );
    expect(liveResult).toMatchSnapshot();
  });
});

describe('When invoked a LiveResult component, with other values - ad x2', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    currentMatch: {
      flow: {
        sets: [
          {
            p1: 0,
            p2: 1,
          },
          {
            p1: 1,
            p2: 1,
          },
        ],
      },
      result: [
        {
          games: [0, 0, 0],
          name: 'Anna',
        },
        {
          games: [0, 0, 0],
          name: 'Sala',
        },
      ],
    },
    currentGamePoints: {
      points: [
        {
          p1: 0,
          p2: 15,
        },
        {
          p1: 43,
          p2: 44,
        },
      ],
    },
    currentSetGames: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 6,
      },
    ],
    currentMatchSets: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 2,
      },
    ],
    server: true,
    tokens: ['abc', 'cde'],
    user: {user: {playerId: '123cde'}},
  });
  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'updateSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateServer').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatch').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });

  test('Should render LiveResult component', () => {
    const liveResult = render(
      <Provider store={store}>
        <LiveResult navigation={navigation} />
      </Provider>,
    );
    expect(liveResult).toMatchSnapshot();
  });
});

describe('When invoked a LiveResult component, with other values - duce x2', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    currentMatch: {
      flow: {
        sets: [
          {
            p1: 0,
            p2: 1,
          },
          {
            p1: 1,
            p2: 1,
          },
        ],
      },
      result: [
        {
          games: [0, 0, 0],
          name: 'Anna',
        },
        {
          games: [0, 0, 0],
          name: 'Sala',
        },
      ],
    },
    currentGamePoints: {
      points: [
        {
          p1: 0,
          p2: 15,
        },
        {
          p1: 44,
          p2: 44,
        },
      ],
    },
    currentSetGames: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 6,
      },
    ],
    currentMatchSets: [
      {
        p1: 0,
        p2: 1,
      },
      {
        p1: 1,
        p2: 2,
      },
    ],
    server: true,
    tokens: ['abc', 'cde'],
    user: {user: {playerId: '123cde'}},
  });
  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'updateSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateServer').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatch').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });

  test('Should render LiveResult component', () => {
    const liveResult = render(
      <Provider store={store}>
        <LiveResult navigation={navigation} />
      </Provider>,
    );
    expect(liveResult).toMatchSnapshot();
  });
});

describe('When invoked a LiveResult component currentMatch.result = undefined', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    currentMatch: {
      flow: {
        sets: [],
      },
    },
    currentGamePoints: {
      points: [],
    },
    currentSetGames: [],
    currentMatchSets: [],
    server: true,
    tokens: ['abc', 'cde'],
    user: {user: {playerId: '123cde'}},
  });
  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'updateSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchGames').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatchSets').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateServer').mockReturnValueOnce({type: ''});
    jest.spyOn(actions, 'updateMatch').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });

  test('Should render LiveResult component', () => {
    const liveResult = render(
      <Provider store={store}>
        <LiveResult navigation={navigation} />
      </Provider>,
    );
    expect(liveResult).toMatchSnapshot();
  });
});
