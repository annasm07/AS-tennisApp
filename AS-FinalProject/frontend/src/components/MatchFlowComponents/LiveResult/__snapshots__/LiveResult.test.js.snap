// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a LiveResult component Should render LiveResult component 1`] = `
<View
  style={
    Object {
      "height": 130,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "margin": 15,
        "marginBottom": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#7A7A7A",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Anna
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Sala
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          Game
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 15,
        "marginHorizontal": 15,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7A7A7A",
        }
      }
    >
      Invalid Date
    </Text>
    <Text
      style={Object {}}
    >
      Best of 3 Sets
    </Text>
  </View>
</View>
`;

exports[`When invoked a LiveResult component currentMatch.result = undefined Should render LiveResult component 1`] = `
<Text>
  ...loading...
</Text>
`;

exports[`When invoked a LiveResult component, with other values - ad Should render LiveResult component 1`] = `
<View
  style={
    Object {
      "height": 130,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "margin": 15,
        "marginBottom": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#7A7A7A",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Anna
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          40
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Sala
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          Ad
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 15,
        "marginHorizontal": 15,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7A7A7A",
        }
      }
    >
      Invalid Date
    </Text>
    <Text
      style={Object {}}
    >
      Best of 3 Sets
    </Text>
  </View>
</View>
`;

exports[`When invoked a LiveResult component, with other values - ad x2 Should render LiveResult component 1`] = `
<View
  style={
    Object {
      "height": 130,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "margin": 15,
        "marginBottom": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#7A7A7A",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Anna
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          40
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Sala
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          Ad
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 15,
        "marginHorizontal": 15,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7A7A7A",
        }
      }
    >
      Invalid Date
    </Text>
    <Text
      style={Object {}}
    >
      Best of 3 Sets
    </Text>
  </View>
</View>
`;

exports[`When invoked a LiveResult component, with other values - duce x2 Should render LiveResult component 1`] = `
<View
  style={
    Object {
      "height": 130,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "margin": 15,
        "marginBottom": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#7A7A7A",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Anna
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          40
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Sala
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          40
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 15,
        "marginHorizontal": 15,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7A7A7A",
        }
      }
    >
      Invalid Date
    </Text>
    <Text
      style={Object {}}
    >
      Best of 3 Sets
    </Text>
  </View>
</View>
`;

exports[`When invoked a LiveResult component, with other values - game Should render LiveResult component 1`] = `
<View
  style={
    Object {
      "height": 130,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "margin": 15,
        "marginBottom": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#7A7A7A",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Anna
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          40
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Sala
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          40
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 15,
        "marginHorizontal": 15,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7A7A7A",
        }
      }
    >
      Invalid Date
    </Text>
    <Text
      style={Object {}}
    >
      Best of 3 Sets
    </Text>
  </View>
</View>
`;

exports[`When invoked a LiveResult component, with other values - game after ad Should render LiveResult component 1`] = `
<View
  style={
    Object {
      "height": 130,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "margin": 15,
        "marginBottom": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#7A7A7A",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Anna
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          40
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Sala
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "marginRight": 15,
            }
          }
        >
          Game
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
        <Text>
           
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 15,
        "marginHorizontal": 15,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7A7A7A",
        }
      }
    >
      Invalid Date
    </Text>
    <Text
      style={Object {}}
    >
      Best of 3 Sets
    </Text>
  </View>
</View>
`;
